src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,365 @@
+  const history = useHistory()
+  const location = useLocation()
+  const appDispatch = useAppDispatch()
+  const {
+    state: { isConnected, isDemoWallet },
+    dispatch,
+  } = useWallet()
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    appDispatch(marketApi.endpoints.findByAssetId.initiate(foxEthLpAssetId))
+  }, [appDispatch])
+
+  const foxEthMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, foxEthLpAssetId),
+  )
+
+  const opportunityId = useMemo(
+    () =>
+      toOpportunityId({
+        assetNamespace: 'erc20',
+        assetReference: ETH_FOX_STAKING_EVERGREEN_CONTRACT,
+        chainId: assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId,
+      }),
+    [assetAccountId],
+  )
+
+  const stakingOpportunityByIdFilter = useMemo(() => {
+    const userStakingId = assetAccountId
+      ? serializeUserStakingId(assetAccountId, opportunityId)
+      : undefined
+    return { userStakingId, opportunityId }
+  }, [assetAccountId, opportunityId])
+
+  const opportunity = useAppSelector(state =>
+    selectEarnUserStakingOpportunityByOpportunityId(state, stakingOpportunityByIdFilter),
+  )
+
+  const { getPeriodFinish } = useFoxFarming(ETH_FOX_STAKING_EVERGREEN_CONTRACT)
+
+  const {
+    data: periodFinish,
+    isLoading: isGetPeriodFinishQueryLoading,
+    isFetching: isGetPeriodFetchingLoading,
+  } = useQuery({
+    queryKey: ['getPeriodFinish'],
+    queryFn: () => getPeriodFinish(),
+  })
+
+  const isGetPeriodFinishLoading = useMemo(
+    () => isGetPeriodFinishQueryLoading || isGetPeriodFetchingLoading,
+    [isGetPeriodFinishQueryLoading, isGetPeriodFetchingLoading],
+  )
+
+  const fetchOpportunityData = useCallback(async () => {
+    if (!foxEthMarketData) return
+
+    const chainId = assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId
+
+    await fetchAllOpportunitiesIdsByChainId(appDispatch, chainId)
+    appDispatch(
+      opportunitiesApi.endpoints.getOpportunityMetadata.initiate(
+        [
+          {
+            opportunityId,
+            defiType: DefiType.Staking,
+            defiProvider: DefiProvider.EthFoxStaking,
+          },
+        ],
+        { forceRefetch: true },
+      ),
+    )
+
+    if (assetAccountId) {
+      appDispatch(
+        opportunitiesApi.endpoints.getOpportunityUserData.initiate(
+          [
+            {
+              opportunityId,
+              accountId: assetAccountId,
+              defiType: DefiType.Staking,
+              defiProvider: DefiProvider.EthFoxStaking,
+            },
+          ],
+          { forceRefetch: true },
+        ),
+      )
+    }
+
+    return true
+  }, [assetAccountId, appDispatch, foxEthMarketData, opportunityId])
+
+  useEffect(() => {
+    queryClient.invalidateQueries({
+      queryKey: ['fetchOpportunityData'],
+    })
+  }, [assetAccountId, queryClient])
+
+  const { isLoading: isOpportunityQueryLoading, isFetching: isOpportunityFetching } = useQuery({
```

Comment:
```suggestion
  const { isLoading: isOpportunityDataLoading, isFetching: isOpportunityDataFetching } = useQuery({
```

Reply:
Done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)